develop/tutorials/articles/140-application-security/01-application-permissions/03-registering-permissions.markdown

```diff
@@ -49,6 +53,77 @@ or group template (site template). These are used internally for those objects,
 so you'd only use them if you were customizing functionality for creating and
 deleting them. 
 
+Now you're ready to register your entities with the permissions service. 
+
+## Registering Entities to the Permissions Service
+
+The permissions service that's running needs to know about your entities and how
+to check permissions for them. This requires creating a permissions registrar
+class. 
+
+1.  In your service bundle, create a package that by convention ends in
+    `internal.security.permission.resource`. For example, the Blogs
+    application's package is named
+    `com.liferay.blogs.internal.security.permission.resource`. 
+
+2.  Create a class in this package called `[Entity
+    Name]ModelResourcePermissionRegistrar`. For example, the Blogs application's
+    class is named `BlogsEntryModelResourcePermissionRegistrar`. 
+
+3.  This class is a component class that requires overriding the `activate`
+    method to register the permissions logic you want for your entities. For
+    example, this is how the Blogs application registers its permissions: 
+
+        @Component(immediate = true)
+        public class BlogsEntryModelResourcePermissionRegistrar {
+
+            @Activate
+            public void activate(BundleContext bundleContext) {
+                Dictionary<String, Object> properties = new HashMapDictionary<>();
+
+                properties.put("model.class.name", BlogsEntry.class.getName());
+
+                _serviceRegistration = bundleContext.registerService(
+                    ModelResourcePermission.class,
+                    ModelResourcePermissionFactory.create(
+                        BlogsEntry.class, BlogsEntry::getEntryId,
+                        _blogsEntryLocalService::getEntry, _portletResourcePermission,
+                        (modelResourcePermission, consumer) -> {
+                            consumer.accept(
+                                new StagedModelPermissionLogic<>(
+                                    _stagingPermission, BlogsPortletKeys.BLOGS,
+                                    BlogsEntry::getEntryId));
+                            consumer.accept(
+                                new WorkflowedModelPermissionLogic<>(
+                                    _workflowPermission, modelResourcePermission,
+                                    BlogsEntry::getEntryId));
+                        }),
+                    properties);
+            }
+
+            @Deactivate
+            public void deactivate() {
+                _serviceRegistration.unregister();
+            }
+
+            @Reference
+            private BlogsEntryLocalService _blogsEntryLocalService;
+
+            @Reference(target = "(resource.name=" + BlogsConstants.RESOURCE_NAME + ")")
+            private PortletResourcePermission _portletResourcePermission;
```

Comment:
This is how they should reference PortletResourcePermissions and ModelResourcePermissions in component classes.  In the case of ModelResourcePermissions they should use a filter matching the "model.class.name" property they set above.